pox-locking/src/events.rs

```diff
@@ -378,7 +440,9 @@ fn create_event_info_data_code(
                         unlock-burn-height: {until_burn_height},
                         ;; optional PoX address tuple.
                         ;; equal to args[3].
-                        pox-addr: {pox_addr}
+                        pox-addr: {pox_addr},
+                        ;; Get end cycle ID
+                        end-cycle-id: (burn-height-to-reward-cycle {until_burn_height}),
```

Comment:
While testing I realized `burn-height-to-reward-cycle` doesn't expect an optional so when passing `until_burn_height` (which is an optional), it leads to a parsing issue and doesn't spit out an event for delegate-stx....just to prove this to myself, I used below (which is not proper at all given the default needs to be something else)
` until_burn_height_val = &args[2].clone().expect_optional().map_or(Value::UInt(0), |v| v.unwrap()),
`

So, if we want `end-cycle-id` to be present in all events, I'm not sure how else it can be calculated so we don't rely on an optional? 
Unless `until_burn_height` is optional for fork/update reasons but in current Stack we're always guaranteed to have Some value.

Reply:
Great catch! I think we'll need to return an optional for `end-cycle-id` in this case. I haven't really tested this code, but I think this should work
```suggestion
                        end-cycle-id: (match {until_burn_height} 
                            height (some (burn-height-to-reward-cycle height))
                            none),
```